### PR TITLE
[aievec] Create a separate dialect for AIE1 ops

### PR DESCRIPTION
**Pull request by jamestcl-amd**
This PR creates a separate dialect for AIE1 ops `add/sub/mac/mul`. It doesn't remove the existing AIE1 ops in the AIEVEC dialect to facilitate easier review. The actual migration of `add/sub/mac/mul` ops and tests to the AIE1 aievec can be found at https://github.com/Xilinx/mlir-aie/pull/1633.

- The header and source files of the AIE1 dialect and ops are structured as follows:
```
include/aie/Dialect/AIEVec/AIE1/IR/AIEVecAIE1Dialect.h
include/aie/Dialect/AIEVec/AIE1/IR/AIEVecAIE1Dialect.td
include/aie/Dialect/AIEVec/AIE1/IR/AIEVecAIE1Ops.h
include/aie/Dialect/AIEVec/AIE1/IR/AIEVecAIE1Ops.td

lib/Dialect/AIEVec/IR/AIE1/AIEVecAIE1Ops.cpp
```
- The name of the dialect is `aievec_aie1` in the `namespace ::xilinx::aievec::aie1`.
- Add a roundtrip test for the AIE1 ops at  `test/dialect/AIEVec/AIE1/roundtrip.mlir`. For example, an add op in IR is `%0 = aievec_aie1.add %A, %B : vector<8xi32>, vector<8xi32>, vector<8xi32>`.

### CODE REVIEW

**Review by david-vc — APPROVED** [review]
LGTM!

File: lib/Dialect/AIEVec/IR/AIE1/AIEVecAIE1Ops.cpp [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+//===----------------------------------------------------------------------===//
+// AIEVecAIE1Dialect
+//===----------------------------------------------------------------------===//
+
+void AIEVecAIE1Dialect::initialize() {
+  // registerTypes();
+  //   addAttributes<
+  // #define GET_ATTRDEF_LIST
+  // #include "aie/Dialect/AIEVec/IR/AIEVecAttributes.cpp.inc"
+  //       >();
```
Comment (jsetoain) [inline lib/Dialect/AIEVec/IR/AIE1/AIEVecAIE1Ops.cpp:40]: Do we need those comments?

**Review by jsetoain — APPROVED** [review]
Thanks, James! Great job!

File: lib/Dialect/AIEVec/IR/AIE1/AIEVecAIE1Ops.cpp (hunk 1 above)
Comment (jamestcl-amd) [reply lib/Dialect/AIEVec/IR/AIE1/AIEVecAIE1Ops.cpp:40]: Will clean up the left over comments in next PR :)

**Pull request merged**